evalml/model_understanding/force_plots.py

```diff
@@ -0,0 +1,100 @@
+                                              cls_dict["expected_value"],
+                                              matplotlib=matplotlib)
+    return shap_plots
+
+
+def force_plot(pipeline, rows_to_explain, training_data, y):
+    """ Function to generate the data required to build a force plot.
+
+    Args:
+        pipeline (PipelineBase): The pipeline to generate the force plot for.
+        rows_to_explain (list(int)): A list of the indices of the training_data to explain.
+        training_data (pandas.DataFrame): The data used to train the pipeline.
+        y (pandas.Series): The target data.
+
+    Returns:
+        list(dict()): list of dictionaries where each dict contains force plot data.  Each dictionary
+            entry represents the explanations for a single row.
+            e.x. For single row binary force plots:
+                    [{'malignant': {'expected_value': 0.37,
+                                    'feature_names': ['worst concave points', 'worst perimeter', 'worst radius'],
+                                    'shap_values': [0.09, 0.09, 0.08],
+                                    'plot': AdditiveForceVisualizer}]
+                 For two row binary force plots:
+                    [{'malignant': {'expected_value': 0.37,
+                                    'feature_names': ['worst concave points', 'worst perimeter', 'worst radius'],
+                                    'shap_values': [0.09, 0.09, 0.08],
+                                    'plot': AdditiveForceVisualizer},
+                    {'malignant': {'expected_value': 0.29,
+                                    'feature_names': ['worst concave points', 'worst perimeter', 'worst radius'],
+                                    'shap_values': [0.05, 0.03, 0.02],
+                                    'plot': AdditiveForceVisualizer}]
+
+    Raises:
+        TypeError: if rows_to_explain is not a list.
+        TypeError: if all values in rows_to_explain aren't integers.
+    """
+
+    if not isinstance(rows_to_explain, list):
+        raise TypeError("rows_to_explain should be provided as a list of row index integers!")
+    if not all([isinstance(x, int) for x in rows_to_explain]):
+        raise TypeError("rows_to_explain should only contain integers!")
+
+    explanations = []
+    prediction_explanations = explain_predictions(pipeline, training_data, y, rows_to_explain,
+                                                  top_k_features=len(training_data.columns), include_shap_values=True,
+                                                  output_format="dict")
+    row_explanations = prediction_explanations["explanations"]
+    for row_explanation in row_explanations:
+        row_exp = row_explanation["explanations"]
+        row_exp_dict = {}
+        for cls_exp in row_exp:
+            cls = cls_exp["class_name"] if cls_exp["class_name"] is not None else "regression"
```

Comment:
So, according to this logic, if the `"class_name"` comes back as `None` here, and we set `cls` to `"regression"`, doesn't that mean the assignment to `row_exp_dict[cls]` will be overwriting the same key for every loop in this `for` loop?

Reply:
The possible values for the key "class_name" are None, the name of the positive class ("malignant" in these tests), or the names of the different classes ["class_0", "class_1", "class_2"].  the row_exp variable is a list that contains either 1 explanation for regression type problems, 1 explanation for binary class problems (the positive class) or multiple explanations for multi-class problems.  If the row for the regression type problems had more than one "class" to explain then yes, it would continuously overwrite that key, but in this case it should be good.